pkg/build/builder/daemonless.go

```diff
@@ -51,14 +51,21 @@ func pullDaemonlessImage(sc types.SystemContext, store storage.Store, imageName
 		return fmt.Errorf("error parsing image name %s: %v", ref, err)
 	}
 	if ref.Registry == "" {
+		// Due to https://github.com/containers/image/issues/572, buildah will default
+		// the registry to docker.io. If docker.io is not in the registry search path
+		// (set in /etc/registries.conf), buildah may be able to pull the image,
+		// but will not be able to use it when it runs build commands.
 		glog.V(2).Infof("defaulting registry to docker.io for image %s", imageName)
```

Comment:
i don't think we should print this message.  at best you might want to log a "no registry specified" message, but we have no idea what the defaulting behavior will be.

Reply:
I kept this in the event https://github.com/openshift/origin/pull/21864 lands - tests there explicitly look for this message.

Reply:
ok, we'll probably need a couple more turns of the crank (but let's let 21864 land as is) because ultimately i'd want to get rid of this.